src/packages/emmett-mongodb/src/eventStore/mongoDBEventStore.ts

```diff
@@ -82,22 +98,23 @@ export class MongoDBEventStore implements EventStore<number> {
     }
 
     assertExpectedVersionMatchesCurrent(
-      stream.events.length,
+      stream.metadata.streamPosition,
       expectedStreamVersion,
       MongoDBEventStoreDefaultStreamVersion,
     );
 
     return {
+      // TODO: remove `.slice`?
```

Comment:
**COMMENT:** Slice is needed when we support reading a part of stream, see, e.g. how PostgreSQL filter implementation: https://github.com/event-driven-io/emmett/blob/97519d0b561943a4a096f44cf95c4c8fd1393de8/src/packages/emmett-postgresql/src/eventStore/schema/readStream.ts#L40. Here we could either use project from mongo, to just read the subset of array, or filter it with slice.

Reply:
I think I'll do this at the DB level since it supports it. Would `currentStreamPosition` remain as the current version of the whole stream or should it reflect what was returned based on `expectedStreamVersion`

Reply:
Good question! I think I'd still return the current version of the whole stream, as this is meant to be used later in the optimistic concurrency check. Plus, the name suggests that it should be the current stream position, so I think that's worth following the least surprise principle. If someone wants to read the last position of read events, they'll be able to do it from event metadata. Maybe in the future, it'd be worth adding additional property if it was used more often.